### DISCUSSION TITLE
[Feature Request] Projection-event applier lambda functions

### DISCUSSION THREAD

**Discussion by malscent** [JasperFx/marten#2803] [General]
At times, when creating modular software, it would be useful to be able to add events and event appliers for existing projections from external libraries.

An example:

If i have ProjectionA in ClassLibA with methods to apply all events within ClassLibA, if I want to add an optional feature by including another library/module (ClassLibB), the events for the optional feature must reside in ClassLibA, and ProjectionA must contain Apply methods for them.

This would break the separation of concerns and remove the benefits of breaking optional feature into a module/library.

I propose that we allow registration of EventHandlers for specific Projections and Events. 
[Here is a simplistic gist example](https://gist.github.com/malscent/3a3d02b7bb2c35bd22a95fdf16540e39)

**Comment by jeremydmiller** [comment]
This is actually *technically* possible today, just there's no documentation or sample to show you how.

**Comment by mysticmind** [comment]
Moving this to discussion until we attempt to address the docs.
